balancer/rls/internal/config.go

```diff
@@ -62,9 +63,10 @@ type lbConfig struct {
 	staleAge             time.Duration
 	cacheSizeBytes       int64
 	defaultTarget        string
-	cpName               string
-	cpTargetField        string
-	cpConfig             map[string]json.RawMessage
+
+	childPolicyName        string
+	childPolicyConfig      json.RawMessage
```

Comment:
How will `childPolicyConfig` be used?
Will you need to add the extra field? Don't want to keep this as a map?

Reply:
When the RLS policy receives an RLS response with a set of targets, it will create one child policy per target. For each of the child policies to be created, the base config will be appended with the actual target for which the child policy is built.

I changed it back and forth a couple of time as I was changing other stuff as well. I think map works better. But I will change it to have the extra field in the map always (created with the dummy value). Whenever a child policy needs to be created, this map entry will be replaced with the actual target.

Reply:
I think whether this map has the dummy value or not doesn't really matter, because you are going to set the value in the map anyway.
(Hmm, you will also need to copy the map I assume? In that case, having the dummy value means you need to also copy the extra value? That's a minor problem I think).

Reply:
We wouldn't have to copy the map. We will marshal it into JSON before passing it to the `ParseConfig` method of the child policy.